woocommerce/woocommerce-ios
# [IAP] Hide hardcoded features when remote flag off

Closes: #9909 


## Description

We're adding the ability to purchase the Woo Express essential plan via in-app purchase, and to ship more quickly, we get the feature 

We previously added a remote flag to allow us to invalidate the hardcoded data regarding Woo Express plans.

With this PR, we use it in the UpgradesView. We assume it’s `true` until we fetch `false`, at which point we hide the feature details, and replace them with a button to open the express page in a web view. This is a fallback only, and we’ll remove it once we can fetch the feature details from an API.


## Testing instructions


1. Set up Proxyman with a breakpoint on `https://public-api.wordpress.com/wpcom/v2/mobile/feature-flags`
2. Launch the app and switch to a free trial site
3. Tap `Upgrade` on the banner on the dashboard
4. Observe that the feature benefits show, and proxyman hits your breakpoint
5. Edit the response to make the `woo_hardcoded_plan_upgrade_details_milestone_1_are_accurate` `false`
6. Observe that the feature benefits are hidden, and there's a link to view the plans on the web instead.

## Screenshots



---
